packages/gatsby/src/utils/babel-loader-helpers.js

```diff
@@ -62,6 +62,7 @@ const prepareOptions = (babel, options = {}, resolve = require.resolve) => {
           resolve(`./babel/babel-plugin-remove-api`),
           {
             apis: [`getServerData`, `config`],
+            scope: `src/pages`,
```

Comment:
Assumes a relative directory or page path is sufficient for cases where `babel-plugin-remove-api` would be used